subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/fixture/AbstractJvmFailFastIntegrationSpec.groovy

```diff
@@ -0,0 +1,242 @@
+        result.testClass('pkg.OtherTest').assertTestCount(0, 0, 0)
+    }
+
+    def "stop test execution with --fail-fast"() {
+        given:
+        withBuildFile()
+        withFailingTest()
+        withNonfailingTest()
+        def testExecution = server.expectConcurrentAndBlock(2, FAILED_RESOURCE, OTHER_RESOURCE)
+
+        when:
+        def gradleHandle = executer.withTasks('test', '--fail-fast').start()
+        testExecution.waitForAllPendingCalls()
+
+        then:
+        testExecution.release(FAILED_RESOURCE)
+        sleep(1000)
+        testExecution.release(OTHER_RESOURCE)
+        gradleHandle.waitForFailure()
+        def result = new DefaultTestExecutionResult(testDirectory)
+        result.testClass('pkg.FailedTest').assertTestFailed('failTest', Matchers.anything())
+        result.testClass('pkg.OtherTest').assertTestCount(0, 0, 0)
+    }
+
+    @Unroll
+    def "ensure fail fast with forkEvery #forkEvery, maxWorkers #maxWorkers, omittedTests #testOmitted"() {
+        given:
+        withBuildFile(maxWorkers, forkEvery)
+        withFailingTest()
+        def otherResources = withNonfailingTests(testOmitted)
+        def testExecution = server.expectOptionalAndBlock(maxWorkers, ([FAILED_RESOURCE ] + otherResources).grep() as String[])
+
+        when:
+        def gradleHandle = executer.withTasks('test', '--fail-fast').start()
+        testExecution.waitForAllPendingCalls()
+
+        then:
+        testExecution.release(FAILED_RESOURCE)
+        sleep(1000)
+        testExecution.releaseAll()
+        gradleHandle.waitForFailure()
+        def result = new DefaultTestExecutionResult(testDirectory)
+        result.testClass('pkg.FailedTest').assertTestFailed('failTest', Matchers.anything())
+        if (result.getTotalNumberOfTestClassesExecuted() != 1) {
+            result.testClassStartsWith("pkg.OtherTest").assertTestCount(0, 0, 0)
+        }
+
+        where:
+        forkEvery | maxWorkers | testOmitted
+        0         | 1          | 1
+        1         | 1          | 1
+        2         | 1          | 1
+        0         | 2          | 5
+        1         | 2          | 5
+        2         | 2          | 5
+    }
+
+    def "fail fast console output shows failure"() {
+        given:
+        withBuildFile()
+        withFailingTest()
+        withNonfailingTest()
+        def testExecution = server.expectConcurrentAndBlock(2, FAILED_RESOURCE, OTHER_RESOURCE)
+
+        when:
+        def gradleHandle = executer.withTasks('test', '--fail-fast').start()
+        testExecution.waitForAllPendingCalls()
+
+        then:
+        testExecution.release(FAILED_RESOURCE)
+        gradleHandle.waitForFailure()
+        assert gradleHandle.standardOutput.matches(/(?s).*pkg\.FailedTest.*failTest.*FAILED.*java.lang.RuntimeException at FailedTest.java.*/)
+        assert !gradleHandle.standardOutput.contains('pkg.OtherTest')
+    }
+
+    void withBuildFile(int maxWorkers = MAX_WORKERS, int forkEvery = 0) {
```

Comment:
Might be worth to move those fixtures into a separate class. Add a constructor to the new class for passing in the `TestFile` instance so that you can create new `file` references.

Reply:
done (but it took a little more than just the `TestFile`)